src/modules/i18n/tests/useTranslations.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { describe, expect, test, vi } from "vitest";
+import {
+  type I18nContext,
+  useI18nContext,
+} from "~/modules/i18n/context/I18nContext";
+import { useTranslations } from "~/modules/i18n/hooks/useTranslations";
+import type { LanguageTag } from "~/modules/i18n/i18n";
+import type { Locale } from "~/modules/i18n/utils/locales";
+
+vi.mock("~/modules/i18n/context/I18nContext");
+
+describe("useTranslations", () => {
+  test.each<{
+    test: string;
+    languageTag: LanguageTag;
```

Comment:
```suggestion
    languageTag: LanguageTag | string;
```
...so that language tags which are not included in the configured language tags don't cause typescript errors, like "de-DE".